src/main/java/uk/ac/ebi/ega/egafuse/runner/CommandLineOptionParser.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ega.egafuse.runner;
+
+import joptsimple.OptionParser;
+
+public class CommandLineOptionParser {
+
+    public static final String OPTIONS_HELP = "h";
+
+    private static final OptionParser optionParser = buildParser();
+
+    private static OptionParser buildParser() {
+        OptionParser parser = new OptionParser();
```

Comment:
Please go through the documentation of Jopts. You can configure option dependency and set that an option will only be  required in another is present or that if username is present, bearer token or credential file should not be present... depending on the limitations that you want to enforce in the command line options. Also, you can define default values for things like parallel connections.

Also at the moment I can't see how we can need so many option to provide the credentials. I understant that it is good to provide them in a file but I can't see how the application won't be able to get the renew token by itself later.

Reply:
Did the changes you have suggested.

Reply:
Also removed the max connection from here will add in 2nd part put in cofiguration

Reply:
I do see the point in having the max connection as an optional parameter with a default value of 1. Usually if we accept command line parameters we should avoid having the user to also modify a separate configuration file. Please add the option again.

Reply:
Done